### PR TITLE
tests: refactor systests into separate modules w/ pytest fixtures

### LINKED ISSUE #391: Split `tests/system.py` into separate modules, refactor to use `pytest` fixtures.
The `tests/system.py` module is huge:
```bash
$ wc -l tests/system.py
1504 tests/system.py
```
and has complicated, idiosyncratic setup / teardown (it long predates our use of `pytest` idioms).

For reference, see https://github.com/googleapis/python-storage/pull/476.

### PR DESCRIPTION
**Pull request by tseaver**
Closes #391.

### CODE REVIEW

**Comment by tseaver** [comment]
Flaky sample error reported as #394.

**Comment by tseaver** [comment]
#394 strikes again.  I'll retry here after getting #395 merged.

**Review by chrisrossi — CHANGES_REQUESTED** [review]

File: tests/system/test_data_api.py [hunk 1]
```diff
@@ -0,0 +1,401 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import datetime
+import operator
+
+import pytest
+
+COLUMN_FAMILY_ID1 = "col-fam-id1"
```
[the remark this answers is not in the document]
Comment (tseaver) [reply tests/system/test_data_api.py:20]: It is an oddity of the back-end that row keys, column names, and cell values must be bytes (at the protobuf level), while column family IDs are strings.

File: tests/system/test_data_api.py [hunk 2]
```diff
@@ -0,0 +1,401 @@
...
+
+def test_table_read_rows_filter_millis(data_table):
+    from google.cloud.bigtable import row_filters
+
+    end = datetime.datetime.now()
+    start = end - datetime.timedelta(minutes=60)
+    timestamp_range = row_filters.TimestampRange(start=start, end=end)
+    timefilter = row_filters.TimestampRangeFilter(timestamp_range)
+    row_data = data_table.read_rows(filter_=timefilter)
+    row_data.consume_all()
```
[the remark this answers is not in the document]
Comment (tseaver) [reply tests/system/test_data_api.py:70]: The test is added in https://github.com/googleapis/google-cloud-python/pull/5002.  It feels like a "smoke test" for the expectation that timestamps passed in as filters get rounded down / up to the nearest millisecond (the back-end would barf when passed microsecond values (https://github.com/googleapis/google-cloud-python/issues/4932).

File: tests/system/conftest.py [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+@pytest.fixture(scope="session")
+def admin_cluster(admin_instance, admin_cluster_id, location_id, serve_nodes):
+    return admin_instance.cluster(
+        admin_cluster_id, location_id=location_id, serve_nodes=serve_nodes,
+    )
+
+
+@pytest.fixture(scope="session")
+def admin_instance_populated(admin_instance, admin_cluster, in_emulator):
+    if not in_emulator:
```
[the remark this answers is not in the document]
Comment (tseaver) [reply tests/system/conftest.py:114]: 54bf4c1

File: tests/system/conftest.py [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+def data_instance_populated(
+    admin_client,
+    data_instance_id,
+    instance_labels,
+    data_cluster_id,
+    location_id,
+    serve_nodes,
+    in_emulator,
+):
+    if not in_emulator:
```
[the remark this answers is not in the document]
Comment (tseaver) [reply tests/system/conftest.py:151]: 54bf4c1

File: tests/system/test_data_api.py [hunk 5]
```diff
@@ -0,0 +1,401 @@
...
+
+    for row_key in row_keys:
+        row = data_table.direct_row(row_key)
+        row.set_cell(COLUMN_FAMILY_ID1, COL_NAME1, CELL_VAL1)
+        row.commit()
+        rows_to_delete.append(row)
+
+    data_table.truncate(timeout=200)
+
+    for row in data_table.read_rows():
```
[the remark this answers is not in the document]
Comment (tseaver) [reply tests/system/test_data_api.py]: That test was originally added in https://github.com/googleapis/google-cloud-python/pull/5360:  I just ported it.

Assuming we don't have race conditions with other CI runs, I would expect there to be no rows in the table after a call to `truncate`.  4b11ce8 changes the assertion accordingly.

File: tests/system/test_data_api.py [hunk 6]
```diff
@@ -0,0 +1,401 @@
...
+        b"row_key_1",
+        b"row_key_2",
+        b"row_key_3",
+        b"row_key_4",
+        b"sample_row_key_1",
+        b"sample_row_key_2",
+    ]
+
+    rows = []
+    for row_key in row_keys:
```
[the remark this answers is not in the document]
Comment (tseaver) [reply tests/system/test_data_api.py]: bec068c

**Review by chrisrossi — APPROVED** [review]
LGTM, thanks!

**Pull request merged**